lib/workers/repository/extract/extract-fingerprint-config.ts

```diff
@@ -0,0 +1,80 @@
+import type { RenovateConfig } from '../../../config/types';
+import { get, getManagerList } from '../../../modules/manager';
+import type { WorkerExtractConfig } from '../../types';
+
+export interface FingerprintExtractConfig {
+  managerList: string[];
+  managers: WorkerExtractConfig[];
+}
+
+function getFilteredManagerConfig(
+  config: RenovateConfig,
+  manager: string
+): WorkerExtractConfig {
+  // the type here is not completely correct because there isn't any defined type only for managers
+  const managerConfig: WorkerExtractConfig = config[manager] as any;
+  const language = get(manager, 'language');
+  // the type here is not completely correct because there isn't any defined type only for languages
+  const languageConfig = (
+    language ? (config[language] ? config[language] : {}) : {}
+  ) as WorkerExtractConfig;
+  const filteredConfig = {} as WorkerExtractConfig;
+
+  filteredConfig.manager = manager;
+  filteredConfig.npmrc = config.npmrc;
+  filteredConfig.npmrcMerge = config.npmrcMerge;
+
+  //  non-mergeable so manager level config is given preference
+  filteredConfig.enabled = managerConfig?.enabled ?? config.enabled;
+  filteredConfig.ignorePaths = managerConfig?.ignorePaths ?? [];
+  filteredConfig.includePaths =
+    managerConfig?.includePaths ?? config.includePaths ?? [];
+  filteredConfig.skipInstalls =
+    managerConfig?.skipInstalls ?? config.skipInstalls;
+
+  // mergeable so manager level config is given preference
+  filteredConfig.registryAliases = {
+    ...config.registryAliases,
+    ...(managerConfig?.registryAliases ?? {}),
+  };
+  filteredConfig.fileMatch = [...(managerConfig?.fileMatch ?? [])].concat(
+    ...(languageConfig?.fileMatch ?? [])
+  );
+
+  return filteredConfig;
```

Comment:
@rarkins do we apply package rules before extract? so we need to consider them here too, as they can change the config?

Reply:
rarkins excluded `packageRules` from fingerprint once before in this PR https://github.com/renovatebot/renovate/pull/18148 so I think we are okay with this

Reply:
I'm pretty sure we don't apply them